docs/content/deployments.md

```diff
@@ -315,4 +315,55 @@ variables.
\ No newline at end of file
+For more information see the [Resource Utilization section on the Policy Performance page](../policy-performance#resource-utilization).
+
+## Operational Readiness and Failure Modes
+
+Depending on how you deploy OPA, it may or may not have policies available as soon as it starts up.  If OPA starts making decisions without any policies, it will return `undefined` as an answer to all policy queries.  This can be problematic because even though OPA returns a response, it has not actually returned the decision dictated by policy.
+
+For example, without loading any policies into OPA whatsoever, a policy query will return the answer `undefined`, which via the HTTP API is represented as an empty JSON object `{}`.
+
+```
+$ opa run -s
+$ curl localhost:8181/v1/data/foo/bar
+{}
+```
+
+In contrast, when policies are loaded, OPA is operationally ready for policy queries, and the answer is defined, the answer is a JSON object of the form `{"result": ...}`
+```
+$ opa run foo.rego -s
+$ curl localhost:8181/v1/data/foo/bar
+{"result": 7}
+```
+
+However, it is possible that even though policies have been loaded the policy response is still `undefined` because the policy makes no decision for the given inputs.
+```
+$ opa run foo.rego -s
+$ curl localhost:8181/v1/data/baz
+{}
+```
+
+Just because OPA has returned an answer for a policy query, that does not indicate that it was operationally ready for that query.  Moreover, the operational readiness of OPA cannot be ascertained from the query response, as illustrated above.  Two issues must therefore be addressed: how to know when OPA is operationally ready for policy queries and how to make a decision before OPA is ready.
+
+### Ensuring Operational Readiness
+The relevance of the discussion above depends on how you have chosen to deploy policies into OPA.
+
+If you deploy policies to OPA on disk (e.g. volume mounting into the OPA container on Kubernetes), then OPA will only start answering policy queries once all the policies are successfully loaded.  In this case, it is impossible for OPA to answer policy queries before it has loaded policy, so the discussion above is a non-issue.
+
+On the other hand, if you use the [Bundle service](management.md#Bundles) OPA will start up without any policies and immediately start downloading a bundle.  But even before the bundle has successfully downloaded, OPA will answer policy queries if asked (which is in every case except the bootstrap case the right thing to do).  For this reason, OPA provides a `/health` [API](rest-api.md#health-api) that verifies that the server is operational and optionally that a bundle has been successfully activated.  As long as no policy queries are routed to OPA until the `/health` API verifies that OPA is operational.  The recommendation is to ensure the `/health` API indicates that OPA is operational before routing policy queries to it.
```

Comment:
`management.md#Bundles` > `../management.md#Bundles`

`rest-api.md#health-api` > `../rest-api.md#health-api`

Reply:
Done

Reply:
Need to remove the `.md` extensions I think.